src/libraries/System.Linq/src/System/Linq/Iterator.cs

```diff
@@ -94,19 +87,21 @@ public IEnumerator<TSource> GetEnumerator()
             /// </summary>
             /// <typeparam name="TResult">The type of the mapped items.</typeparam>
             /// <param name="selector">The selector used to map each item.</param>
-            public virtual IEnumerable<TResult> Select<TResult>(Func<TSource, TResult> selector)
-            {
-                return new SelectEnumerableIterator<TSource, TResult>(this, selector);
-            }
+            public virtual IEnumerable<TResult> Select<TResult>(Func<TSource, TResult> selector) =>
+                new
```

Comment:
Nit: it's a matter of personal preference, but I would rather repeat `new` and constructor arguments over breaking up the expression like this.